src/main/java/org/folio/invoices/utils/HelperUtils.java

```diff
@@ -231,4 +242,83 @@ private static boolean isFieldNotEmpty(Object newObject, Object existedObject, S
   private static boolean isFieldChanged(Object newObject, Object existedObject, String field) throws IllegalAccessException {
     return !EqualsBuilder.reflectionEquals(FieldUtils.readDeclaredField(newObject, field, true), FieldUtils.readDeclaredField(existedObject, field, true), true, existedObject.getClass(), true);
   }
+
+  /**
+   * Transform list of id's to CQL query using 'or' operation
+   * @param ids list of id's
+   * @return String representing CQL query to get records by id's
+   */
+  public static String convertIdsToCqlQuery(List<String> ids) {
+    return StreamEx.of(ids).map(id -> "id==" + id).joining(" or ");
+  }
+
+  /**
+   * Wait for all requests completion and collect all resulting objects. In case any failed, complete resulting future with the exception
+   * @param futures list of futures and each produces resulting object on completion
+   * @param <T> resulting type
+   * @return resulting objects
+   */
+  public static <T> CompletableFuture<List<T>> collectResultsOnSuccess(List<CompletableFuture<T>> futures) {
+    return VertxCompletableFuture.allOf(futures.toArray(new CompletableFuture[0]))
+      .thenApply(v -> futures
+        .stream()
+        // The CompletableFuture::join can be safely used because the `allOf` guaranties success at this step
+        .map(CompletableFuture::join)
+        .filter(Objects::nonNull)
+        .collect(toList())
+      );
+  }
+
+  public static double calculateVoucherAmount(Voucher voucher, List<VoucherLine> voucherLines) {
+
+    CurrencyUnit currency = Monetary.getCurrency(voucher.getSystemCurrency());
+
+    MonetaryAmount amount = voucherLines.stream()
+      .map(line -> Money.of(line.getAmount(), currency))
+      .collect(MonetaryFunctions.summarizingMonetary(currency))
+      .getSum();
+
+    return convertToDouble(amount);
```

Comment:
I realize that convertToDouble wasn't added as part of this PR, but we probably want to add a javadoc comment or rename the method to indicate that it's performing some rounding.  It's easy to screw up calculations if you start rounding in places where you shouldn't, so we want to make sure it's evident that we're not just converting to double here.

Reply:
Renamed